### ISSUE TITLE
KeyError on underwater_fraction

### ISSUE THREAD

**Issue by koen-van-greevenbroek** [PyPSA/pypsa-eur#220]
## Checklist

- [x] I am using the current [`master`](https://github.com/PyPSA/pypsa-eur/tree/master) branch or the latest [release](https://github.com/PyPSA/pypsa-eur/releases). Please indicate.
- [x] I am running on an up-to-date [`pypsa-eur` environment](https://github.com/PyPSA/pypsa-eur/blob/master/envs/environment.yaml). Update via `conda env update -f envs/environment.yaml`.

## Describe the Bug

I am running PyPSA-Eur with a subset of countries (`NO`, `SE`), and during the `prepare_network` rule I get a KeyError on `underwater_fraction`.

The problem happens in `update_transmission_costs` in `add_electricity.py`. There, the `'underwater_fraction'` of DC links is referenced, but if there are no DC links (such as in my case, apparently), `n.links` may not have an `'underwater_fraction'` column. This leads to the KeyError.

It seems like the quick fix is something like a the line
```
    if n.links.loc[dc_b].empty: return
```
so that we don't try to do anything with DC links if there are none. This appears to solve the issue for me, anyway, although I haven't done a lot of testing. If this seems like a good solution, should I submit a one-line pull request? Or is there a better way to handle it?

Cheers,

Koen

**Comment by fneum** [comment]
Yes, this is sensible case handling. Feel free to prepare a short PR to get the credit for this fix. Cheers!
